clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/TransferIndex.scala

```diff
@@ -0,0 +1,29 @@
+package org.broadinstitute.clio.transfer.model
+
+import io.circe.Json
+import org.broadinstitute.clio.util.json.JsonSchemas
+
+/**
+  * This models an index in our database. An index is a table (schema) within our database.
+  */
+sealed trait TransferIndex {
+
+  /**
+    * Encode the index as a url segment for making web service API calls
+    *
+    * @return the url path representation of this index
+    */
+  def urlSegment: String
+
+  def jsonSchema: Json
+}
+
+final case class GvcfIndex() extends TransferIndex {
```

Comment:
Since these don't take parameters, they can be `case object`s

Reply:
I knew there was something I was missing here. I was also thinking of using enums, but wasn't sure where this class was going yet.